core/src/main/java/org/infinispan/commands/tx/PrepareCommand.java

```diff
@@ -189,7 +189,18 @@ public Object getKeyLockOwner() {
 
    @Override
    public boolean hasZeroLockAcquisition() {
-      return false;
+      if (modifications == null || modifications.length == 0) {
+         return false;
+      }
+      boolean hasZeroLock = true;
+      for (WriteCommand wc : modifications) {
+         // If even a single command doesn't have the zero lock acquisition timeout flag, we can't use a zero timeout
+         if (!wc.hasAnyFlag(FlagBitSets.ZERO_LOCK_ACQUISITION_TIMEOUT)) {
+            hasZeroLock = false;
```

Comment:
Could `return false` and avoid the variable.

Reply:
Yes, I had this at some point and undid it. I blame being up late at night :)